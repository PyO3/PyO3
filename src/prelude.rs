//! PyO3's prelude.
//!
//! The purpose of this module is to alleviate imports of many commonly used items of the PyO3 crate
//! by adding a glob import to the top of pyo3 heavy modules:
//!
//! ```
//! # #![allow(unused_imports)]
//! use pyo3::prelude::*;
//! ```

pub use crate::conversion::{FromPyObject, IntoPyObject};
#[allow(deprecated)]
pub use crate::conversion::{IntoPy, ToPyObject};
pub use crate::err::{PyErr, PyResult};
pub use crate::instance::{Borrowed, Bound, Py, PyObject};
pub use crate::marker::Python;
pub use crate::pycell::{PyRef, PyRefMut};
pub use crate::pyclass_init::PyClassInitializer;
pub use crate::types::{PyAny, PyModule};

#[cfg(feature = "macros")]
pub use pyo3_macros::{
    pyclass, pyfunction, pymethods, pymodule, FromPyObject, IntoPyObject, IntoPyObjectRef,
};

#[cfg(feature = "macros")]
pub use crate::wrap_pyfunction;
#[cfg(feature = "macros")]
#[allow(deprecated)]
pub use crate::wrap_pyfunction_bound;

pub use crate::types::any::PyAnyMethods;
pub use crate::types::weakref::PyWeakrefMethods;
