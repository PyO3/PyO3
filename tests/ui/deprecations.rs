#![deny(deprecated)]

use pyo3::prelude::*;

#[pyclass(gc)]
struct DeprecatedGc;

#[pyfunction]
fn function_with_required_after_option(_opt: Option<i32>, _x: i32) {}

fn main() {
    function_with_required_after_option(None, 0);
}
