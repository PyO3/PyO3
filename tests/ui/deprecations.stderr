error: use of deprecated constant `pyo3::impl_::deprecations::CALL_ATTRIBUTE`: use `fn __call__` instead of `#[call]`
  --> tests/ui/deprecations.rs:10:7
   |
10 |     #[call]
   |       ^^^^
   |
note: the lint level is defined here
  --> tests/ui/deprecations.rs:1:9
   |
1  | #![deny(deprecated)]
   |         ^^^^^^^^^^
